itests/src/test/java/fi/metatavu/edelphi/test/ui/base/index/ChangePlanTestsBase.java

```diff
@@ -0,0 +1,242 @@
+  }
+  
+  @Test
+  public void testPlusToPremiumNoCompensation() {
+    testPlanChange(planMocker.getPlanId(1), PLUS_TYPE, PLUS_NAME, null, 3, 100, PREMIUM_TYPE, PREMIUM_NAME, 200, null, null);
+    testPlanChange(planMocker.getPlanId(1), PLUS_TYPE, PLUS_NAME, null, 4, 200, PREMIUM_TYPE, PREMIUM_NAME, 360, null, null);
+  }
+  
+  @Test
+  public void testPlusToPremiumWithCompensation() {
+    OffsetDateTime oldEnds = OffsetDateTime.now().plusDays(30);
+    testPlanChange(planMocker.getPlanId(1), PLUS_TYPE, PLUS_NAME, oldEnds, 3, 100, PREMIUM_TYPE, PREMIUM_NAME, 200, 26.1d, null);
+    testPlanChange(planMocker.getPlanId(1), PLUS_TYPE, PLUS_NAME, oldEnds, 4, 200, PREMIUM_TYPE, PREMIUM_NAME, 360, 26.1d, null);
+  }
+  
+  @Test
+  public void testPremiumToPremium() {
+    OffsetDateTime oldEnds = OffsetDateTime.now().plusDays(30);
+    testPlanChange(planMocker.getPlanId(3), PREMIUM_TYPE, PREMIUM_NAME, oldEnds, 3, 100, PREMIUM_TYPE, PREMIUM_NAME, 200, null, 30);
+    testPlanChange(planMocker.getPlanId(3), PREMIUM_TYPE, PREMIUM_NAME, oldEnds, 4, 200, PREMIUM_TYPE, PREMIUM_NAME, 360, null, 30);
+  }
+  
+  @Test
+  public void testCancelPayment() {
+    Date oldSubscriptionEnd = Date.from(OffsetDateTime.now().plusDays(30).toInstant());
+    Double newPlanPrice = 200d;
+    Double compensation = 26.1d;
+    DateFormat dateFormat = DateFormat.getDateInstance(DateFormat.MEDIUM);
+    String oldEndDateFormatted = oldSubscriptionEnd != null ? dateFormat.format(oldSubscriptionEnd) : null;
+    Double totalPrice = Math.max(newPlanPrice - compensation, 0);
+    int newPlanIndex = 3;
+    String oldPlanName = "Plus";
+    
+    NumberFormat currencyFormatter = NumberFormat.getCurrencyInstance();
+    currencyFormatter.setCurrency(Currency.getInstance("EUR"));
+    String compensationFormatted = compensation != null ? currencyFormatter.format(compensation) : null;
+    String totalFormatted = currencyFormatter.format(totalPrice);
+    updateUserPlan(1l, planMocker.getPlanId(1));
+    updateUserSubscription(1l, "PLUS", null, oldSubscriptionEnd);
+    login(ADMIN_EMAIL);
+   
+    navigate(CHANGEPLAN_PAGE);
+    waitPresent("#currentPlanGenericBlockContent");
+    assertText("#currentPlanGenericBlockContent .currentPlanRow:nth-of-type(1) label", "SUBSCRIPTION LEVEL");
+    assertText("#currentPlanGenericBlockContent .currentPlanRow:nth-of-type(1) span", String.format("Your current subscription level is %s", oldPlanName));
+    assertText("#currentPlanGenericBlockContent .currentPlanRow:nth-of-type(2) span", String.format("Your subscription ends %s", oldEndDateFormatted));
+
+    String compensationText = String.format("Remaining time of your old order will be compensated in your new order. Your compensation will be %s", compensationFormatted);
+    assertText(String.format("#changePlanGenericBlockContent .planFieldContainer:nth-of-type(%d) .planHint", newPlanIndex), compensationText);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "#changePlanGenericBlockContent .planFieldContainer:nth-of-type(%d) .planHint" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)